src/script/components/Modals/PrimaryModal/PrimaryModal.tsx

```diff
@@ -55,6 +55,7 @@ export const PrimaryModalComponent: FC = () => {
   const isModalVisible = currentId !== null;
   const passwordValueRef = useRef<HTMLInputElement>(null);
   const [isFormSubmitted, setIsFormSubmitted] = useState(false);
+  const [isBackupPasswordInvalid, setIsBackupPasswordInvalid] = useState(false);
```

Comment:
With this approach, the initial state is broken - when input is empty, the `isBackupPasswordValid` will be set to `true`, causing an error message to show up.

We can't make an exception for an empty state (e.g `passwordInput === "" ? true : isValidPassword(passwordInput)` cause then we would submit the form with no password) 

I'm not happy with that logic either, I gonna refactor it in this sprint, task - https://wearezeta.atlassian.net/browse/WPB-10932

Reply:
Maybe I'm missing something, but I think we can't do that either. Then the initial state (no value) would cause `isBackupPasswordValid` to be `true`, and we don't want that:

<img width="403" alt="Screenshot 2024-10-18 at 09 37 08" src="https://github.com/user-attachments/assets/33f18b8b-493d-4887-9422-2e046f2addd5">

Reply:
Now I see, missed that isFormSubmitted earlier, Friday vibes 😅 Give me a see, I'm gonna refactor this code real quick